pkg/deploy/controller/generictrigger/factory.go

```diff
@@ -91,11 +89,21 @@ func (c *DeploymentTriggerController) waitForSyncedStore(ready chan<- struct{},
 
 func (c *DeploymentTriggerController) addDeploymentConfig(obj interface{}) {
 	dc := obj.(*deployapi.DeploymentConfig)
+	if len(dc.Spec.Triggers) == 0 || dc.Spec.Paused {
+		return
+	}
 	c.enqueueDeploymentConfig(dc)
 }
 
 func (c *DeploymentTriggerController) updateDeploymentConfig(old, cur interface{}) {
 	dc := cur.(*deployapi.DeploymentConfig)
+	if len(dc.Spec.Triggers) == 0 || dc.Spec.Paused {
+		return
+	}
+	// A periodic relist will send update events for all known configs.
+	if kapi.Semantic.DeepEqual(old, cur) {
```

Comment:
Deep equal is pretty expensive.  Is this the *only* way to solve this?  It's possibly more expensive to do this than to enqueue it.